components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/WriterGroupMessageSource.cs

```diff
@@ -54,18 +54,18 @@ public class WriterGroupMessageTrigger : IMessageTrigger, IDisposable {
                 true : false;
 
         /// <inheritdoc/>
-        public OpcAuthenticationMode AuthenticationMode =>_subscriptions?.First()?.Subscription?.Connection.User != null ?
+        public OpcAuthenticationMode AuthenticationMode =>_subscriptions?.First()?.Subscription?.Connection?.User != null ?
                 OpcAuthenticationMode.UsernamePassword : OpcAuthenticationMode.Anonymous;
 
         /// <inheritdoc/>
-        public string AuthenticationUsername =>_subscriptions?.First()?.Subscription?.Connection.User != null ?
+        public string AuthenticationUsername =>_subscriptions?.First()?.Subscription?.Connection?.User?.Value != null ?
                 _serializer.Deserialize<cred>(_subscriptions?
                     .First()?
                     .Subscription?
-                    .Connection
-                    .User
-                    .Value
-                    .ToJson())
+                    .Connection?
```

Comment:
The condition above makes sure that `_subscriptions.First().Subscription.Connection.User.Value` is not `null` so we can remove the `?` checks up to `ToJson()` here.

Reply:
agreed, thought originally better safe than sorry, but you're right.